samples/cpp/benchmark_app/main.cpp

```diff
@@ -410,11 +412,37 @@ int main(int argc, char* argv[]) {
                     return str;
             };
 
-            if (supported(ov::inference_num_threads.name()) && isFlagSetInCommandLine("nthreads")) {
-                device_config.emplace(ov::inference_num_threads(FLAGS_nthreads));
+            if (isFlagSetInCommandLine("nthreads")) {
+                if (supported(ov::inference_num_threads.name()) || if_multi) {
+                    device_config.emplace(ov::inference_num_threads(FLAGS_nthreads));
+                } else if (if_auto) {
+                    for (auto& device : hardware_devices) {
+                        if (device_config.find(device) == device_config.end()) {
+                            device_config.insert(
+                                ov::device::properties(device, ov::inference_num_threads(FLAGS_nthreads)));
+                        } else {
+                            auto& property = device_config[device].as<ov::AnyMap>();
+                            property.emplace(ov::inference_num_threads(FLAGS_nthreads));
+                            device_config.insert(ov::device::properties(device, property));
+                        }
+                    }
+                }
             }
-            if (supported(ov::affinity.name()) && isFlagSetInCommandLine("pin")) {
-                device_config.emplace(ov::affinity(fix_pin_option(FLAGS_pin)));
+            if (isFlagSetInCommandLine("pin")) {
+                if (supported(ov::affinity.name()) || if_multi) {
+                    device_config.emplace(ov::affinity(fix_pin_option(FLAGS_pin)));
+                } else if (if_auto) {
+                    for (auto& device : hardware_devices) {
+                        if (device_config.find(device) == device_config.end()) {
+                            device_config.insert(
+                                ov::device::properties(device, ov::affinity(fix_pin_option(FLAGS_pin))));
+                        } else {
+                            auto& property = device_config[device].as<ov::AnyMap>();
+                            property.emplace(ov::affinity(fix_pin_option(FLAGS_pin)));
+                            device_config.insert(ov::device::properties(device, property));
```

Comment:
1. Logic for setting looks pretty similar, can we wrap it using some function?
2. Python benchmark_app update is required
3. Smoke sample tests should be updated

Reply:
3. Test cases will be covered by CVS-94119

Reply:
2. Work on Python benchmark_app update.